tsconfig.json

```diff
@@ -0,0 +1,101 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Projects */
+    "incremental": true /* Enable incremental compilation */,
+    // "composite": true,                                /* Enable constraints that allow a TypeScript project to be used with project references. */
+    // "tsBuildInfoFile": "./",                          /* Specify the folder for .tsbuildinfo incremental compilation files. */
+    // "disableSourceOfProjectReferenceRedirect": true,  /* Disable preferring source files instead of declaration files when referencing composite projects */
+    // "disableSolutionSearching": true,                 /* Opt a project out of multi-project reference checking when editing. */
+    // "disableReferencedProjectLoad": true,             /* Reduce the number of projects loaded automatically by TypeScript. */
+
+    /* Language and Environment */
+    "target": "es5" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
```

Comment:
Any reason for us to be transpiling down to ES5? The minimum version supported by zip-it-and-ship-it is [10.18.0](https://github.com/netlify/zip-it-and-ship-it/blob/main/package.json#L102), so we should be fine with `"es6"`? On that note, our linter currently warns you whenever you're trying to use a feature that isn't supported by the version in `engines.node`, which is quite useful — what will be the state of that with the introduction of TypeScript and the build step? Will it be able to know what things will be transpiled _to_ and warn you if there's a compatibility issue?

Reply:
ES5 is the default, I didn't change anything about it. Will update to `es6` 👍